httpcore5/src/test/java/org/apache/hc/core5/net/TestHost.java

```diff
@@ -129,4 +129,31 @@ public void testCreateFromStringInvalid() throws Exception {
         }
     }
 
+    @Test
+    public void testIpv6HostAndPort() throws Exception {
+        final Host host = Host.create("[::1]:80");
+        Assert.assertEquals("::1", host.getHostName());
+        Assert.assertEquals(80, host.getPort());
+    }
+
+    @Test
+    public void testIpv6HostAndPortWithoutBrackets() {
+        try {
+            // ambiguous
+            Host.create("::1:80");
+            Assert.fail("IllegalArgumentException expected");
+        } catch (final URISyntaxException expected) {
+            Assert.assertTrue(expected.getMessage().contains("Expected IPv6 address to be enclosed in brackets"));
+        }
+    }
+
+    @Test
+    public void testIpv6HostWithoutPort() {
+        try {
+            Host.create("::1");
+            Assert.fail("IllegalArgumentException expected");
```

Comment:
this test and the test above it reference the wrong exception type in the failure message -- updating